diff --git a/delete.go b/delete.go
index 84f5b62a..f2ab6959 100644
--- a/delete.go
+++ b/delete.go
@@ -3,7 +3,6 @@
 package main

 import (
-	"errors"
 	"fmt"
 	"os"
 	"path/filepath"
@@ -17,14 +16,13 @@ import (

 func killContainer(container libcontainer.Container) error {
 	_ = container.Signal(unix.SIGKILL, false)
-	for i := 0; i < 100; i++ {
+	for {
 		time.Sleep(100 * time.Millisecond)
 		if err := container.Signal(unix.Signal(0), false); err != nil {
 			destroy(container)
 			return nil
 		}
 	}
-	return errors.New("container init still running")
 }

 var deleteCommand = cli.Command{
