From 6e3d4687c79c19a3bc79201e1b4bf70ec23b488f Mon Sep 17 00:00:00 2001
From: Georgi Sabev <georgethebeatle@gmail.com>
Date: Tue, 6 Feb 2018 13:09:34 +0200
Subject: [PATCH] Remove timeout for runc delete

Signed-off-by: Craig Furman <cfurman@pivotal.io>
---
 delete.go | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/delete.go b/delete.go
index e8e4d547..44f7e0ba 100644
--- a/delete.go
+++ b/delete.go
@@ -17,14 +17,13 @@ import (
 
 func killContainer(container libcontainer.Container) error {
 	_ = container.Signal(unix.SIGKILL, false)
-	for i := 0; i < 100; i++ {
+	for {
 		time.Sleep(100 * time.Millisecond)
 		if err := container.Signal(syscall.Signal(0), false); err != nil {
 			destroy(container)
 			return nil
 		}
 	}
-	return fmt.Errorf("container init still running")
 }
 
 var deleteCommand = cli.Command{
-- 
2.15.0

